GeocodingLambda/utils.ts

```diff
@@ -156,26 +158,36 @@ export const addLocationToDB = (
     isoCountryCode
   }: FlattenedLocation,
   timezoneIdentifier: string
-) =>
-  conn.executeResult(
-    `
-      INSERT INTO location (name, city, country, street, streetNumber, postalCode, latitude, longitude, timezoneIdentifier, isoCountryCode)
-      VALUES (:name, :city, :country, :street, :streetNumber, :postalCode, :latitude, :longitude, :timezoneIdentifier, :isoCountryCode)
-    `,
-    {
-      timezoneIdentifier,
-      latitude,
-      longitude,
-      name,
-      city,
-      country,
-      street,
-      streetNumber,
-      postalCode,
-      region,
-      isoCountryCode
+) => {
+  try {
+    return conn.executeResult(
+      `
+        INSERT IGNORE INTO location (name, region, city, country, street, streetNumber, postalCode, latitude, longitude, timezoneIdentifier, isoCountryCode)
+        VALUES (:name, :region, :city, :country, :street, :streetNumber, :postalCode, :latitude, :longitude, :timezoneIdentifier, :isoCountryCode)
+      `,
+      {
+        timezoneIdentifier,
+        latitude,
+        longitude,
+        name,
+        city,
+        country,
+        street,
+        streetNumber,
+        postalCode,
+        region,
+        isoCountryCode
+      }
+    )
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (e: any) {
+    log.error("Could not insert location into cache: ", { error: e })
+    if (!(e.message.includes("Duplicate entry"))) {
```

Comment:
Nit: Would prefer that we checked an explicit error code instead of a string if possible.